server/views/pages/goal/review/index.njk

```diff
@@ -60,6 +62,13 @@
                 <dd class="govuk-summary-list__value">
                   {{ step.title }}
                 </dd>
+                {% if featureToggles.newCreateGoalRoutesEnabled %}
+                <dd class="govuk-summary-list__actions">
+                  <a class="govuk-link" href="/plan/{{ prisonerSummary.prisonNumber }}/goals/{{ loop.index }}/add-step/{{ step.sequenceNumber }}?mode=edit" data-q="change-goal-{{ loop.index }}-step-{{ step.sequenceNumber }}">
```

Comment:
Need to figure out how to access to the goal number here, `{{ loop.index }}` won't work, as we're in a separate loop for the "Steps" and Goal doesn't have a `sequenceNumber` unlike Steps.

Reply:
`{% set goalLoop = loop %}`